runtime/browser/runtime_download_manager_delegate.h

```diff
@@ -40,6 +51,33 @@ class RuntimeDownloadManagerDelegate
  private:
   friend class base::RefCountedThreadSafe<RuntimeDownloadManagerDelegate>;
 
+  struct DownloadSelectFileParams {
+    DownloadSelectFileParams(content::DownloadItem* item,
+                             const content::DownloadTargetCallback& callback);
+    ~DownloadSelectFileParams();
+
+    content::DownloadItem* item;
+    content::DownloadTargetCallback callback;
+  };
+
+#if defined(OS_LINUX) && !defined(OS_TIZEN)
+  class RuntimeDownloadView : public content::WebContentsObserver {
```

Comment:
agree that UI stuffs should be placed within UIDelegate/NativeAppWindow. I will refactor the code based on this idea. Thanks for your suggestions.